src/getMessageAuthor.ts

```diff
@@ -0,0 +1,97 @@
+import {endGroup, info, isDebug, startGroup, warning} from '@actions/core'
+import {context} from '@actions/github'
+import {OctokitClient, User} from './github/types'
+import {senderFromPayload} from './github/webhook'
+import {EnvironmentVariable} from './input'
+import {SlackClient} from './slack/SlackClient'
+import {MemberWithProfile, MessageAuthor} from './slack/types'
+
+export async function getMessageAuthor(
+  octokit: OctokitClient,
+  slack: SlackClient
+): Promise<MessageAuthor | null> {
+  startGroup('Getting message author')
+
+  try {
+    info('Fetching Slack users')
+    const slackUsers = await slack.getRealUsers()
+
+    if (!slackUsers) {
+      throw new Error(
+        `${EnvironmentVariable.SlackBotToken} does not include "users:read" OAuth scope.`
+      )
+    }
+
+    const githubUser = await getGitHubUser(octokit)
+
+    info(`Finding Slack user by name: ${githubUser.name}`)
+    const matchingSlackUsers = slackUsers.filter(
+      (user): user is MemberWithProfile => {
+        return Boolean(
+          user.profile?.real_name === githubUser.name &&
```

Comment:
Both [GitHub](https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user) and [Slack](https://api.slack.com/scopes/users:read.email) _email_ APIs are separately restrictive, so I didn't want to go down that path. If we find this (naive) algorithm to be problematic, we could consider introducing some optional JSON environment variable that maps GitHub usernames to Slack member IDs (which we would have to maintain).

Reply:
Yeah, I think we go with this and only start maintaining some GitHub->Slack KV structure if we have to.